### ISSUE TITLE
LinAlg error in mix_adapt.r_value.make_r_gaussmix

### ISSUE THREAD

**Issue by mreboud** [pypmc/pypmc#87]
When using `make_r_gaussmix`, the Choleski decomposition fails with the following error:
```
LinAlgError: 42-th leading minor of the array is not positive definite
```

**Comment by fredRos** [comment]
@mreboud Sorry for the late reply, the recent github notifications ended up in my spam folder. Could you give me some context in which case this happens? My guess is that the CHolesky decomposition is invoked in  [this line](https://github.com/pypmc/pypmc/blob/master/pypmc/density/gauss.pyx#L43). 
And it may happen depending on how the covariance matrix is created that the decomp. fails. Did you create it as a sample covariance? Perhaps there were too few samples. From your error message, I infer that you work at least in 42 dimensions.  I have seen it work in this many dimensions but I also know that in around that regime, numerical issues can become more severe and the quadratic scaling of the number of parameters in the covariance matrix makes everything much slower. 

Some idea to remedy that would be to do a sparse or banded matrix taking into account only the most important correlations though I'm not sure what the best approach would be. 

A hack would be to optionally add small values to enforce positive definiteness to work around numerical issues

**Comment by mreboud** [comment]
@fredRos Ahah your answer was also considered as a spam...
You are correct concerning the line of the code. I checked my samples and they looked very weird due to unexpected nonphysical behaviors of the Markov chain. In the end, working with more Markov chains, closer to the best fit point partially solved the issue (I don't see this error message any more).

So I think we can close the issue, I thought that it was due to a bug in the code but it is rather the consequence of using `make_r_gaussmix` on highly non-Gaussian samples.
